event/src/unvalidated/builder.rs

```diff
@@ -0,0 +1,343 @@
+            additional: self.additional,
+        }
+    }
+
+    /// Create builder for data events
+    pub fn data<T>(self, id: Cid, prev: Cid, data: T) -> DataBuilder<T> {
+        DataBuilder {
+            id,
+            prev,
+            data,
+            sep: self.sep,
+            additional: self.additional,
+        }
+    }
+}
+
+/// Builder for unsigned unvalidated events
+pub struct InitBuilder {
+    sep: Option<String>,
+    controllers: Vec<String>,
+    additional: ValueMap,
+}
+
+impl Additional for InitBuilder {
+    fn with_additional(mut self, key: String, value: Value) -> Self {
+        self.additional.insert(key, value);
+        self
+    }
+}
+
+impl Controllers for InitBuilder {
+    fn with_controller(mut self, controller: String) -> Self {
+        self.controllers.push(controller);
+        self
+    }
+}
+
+impl Sep for InitBuilder {
+    fn with_sep(mut self, sep: String) -> Self {
+        self.sep = Some(sep);
+        self
+    }
+}
+
+impl InitBuilder {
+    /// Create unsigned init event without data
+    pub async fn build(self) -> anyhow::Result<init::Payload<()>> {
+        if self.controllers.is_empty() {
+            anyhow::bail!("controllers must not be empty");
+        }
+        let sep = self
+            .sep
+            .ok_or_else(|| anyhow::anyhow!("sep must be specified"))?;
+        let header = init::Header::new(self.controllers, sep, self.additional);
+        let payload = init::Payload::new(header, None);
+        Ok(payload)
+    }
+
+    /// Create builder for init event with data
+    pub fn with_data<T>(self, data: T) -> InitWithDataBuilder<T> {
+        InitWithDataBuilder {
+            data,
+            controllers: self.controllers,
+            sep: self.sep,
+            additional: self.additional,
+        }
+    }
+}
+
+/// Builder for init event with data
+pub struct InitWithDataBuilder<D> {
+    data: D,
+    controllers: Vec<String>,
+    sep: Option<String>,
+    additional: ValueMap,
+}
+
+impl<D> Additional for InitWithDataBuilder<D> {
+    fn with_additional(mut self, key: String, value: Value) -> Self {
+        self.additional.insert(key, value);
+        self
+    }
+}
+
+impl<D> Controllers for InitWithDataBuilder<D> {
+    fn with_controller(mut self, controller: String) -> Self {
+        self.controllers.push(controller);
+        self
+    }
+}
+
+impl<D> Sep for InitWithDataBuilder<D> {
+    fn with_sep(mut self, sep: String) -> Self {
+        self.sep = Some(sep);
+        self
+    }
+}
+
+impl<D> InitWithDataBuilder<D> {
+    /// Build unsigned init event with data
+    pub async fn build(self) -> anyhow::Result<init::Payload<D>> {
+        if self.controllers.is_empty() {
+            anyhow::bail!("controllers must not be empty");
+        }
+        let sep = self
+            .sep
+            .ok_or_else(|| anyhow::anyhow!("sep must be specified"))?;
+        let header = init::Header::new(self.controllers, sep, self.additional);
```

Comment:
I don't think we want to allow arbitrary additional fields into event headers.  This is in fact in exact conflict with a task I have to complete this week: https://linear.app/3boxlabs/issue/WS1-1633/disallow-unexpected-metadata-fields-in-events.

Allowing arbitrary metadata into event headers is dangerous as it restricts our ability to add new header fields in the future, since now we might wind up breaking an application that is using that field name in the header already.